phalcon/mvc/modelinterface.zep

```diff
@@ -167,89 +182,74 @@ interface ModelInterface
+	public static function query(<DiInterface> dependencyInjector = null) -> <CriteriaInterface>;
 
 	/**
-	 * Check whether validation process has generated any messages
+	 * Refreshes the model attributes re-querying the record from the database
 	 */
-	public function validationHasFailed() -> boolean;
+	public function refresh();
 
 	/**
-	 * Returns array of validation messages
+	 * Inserts or updates a model instance. Returning true on success or false otherwise.
 	 */
-	public function getMessages() -> <MessageInterface[]>;
+	public function save() -> boolean;
 
 	/**
-	 * Inserts or updates a model instance. Returning true on success or false otherwise.
+	 * Sets both read/write connection services
 	 */
-	public function save() -> boolean;
+	public function setConnectionService(string connectionService) -> void;
 
 	/**
-	 * Inserts a model instance. If the instance already exists in the persistence it will throw an exception
-	 * Returning true on success or false otherwise.
+	 * Sets the dirty state of the object using one of the DIRTY_STATE_* constants
 	 */
-	public function create() -> boolean;
+	public function setDirtyState(int dirtyState) -> <ModelInterface>;
 
 	/**
-	 * Updates a model instance. If the instance doesn't exist in the persistence it will throw an exception
-	 * Returning true on success or false otherwise.
+	 * Sets the DependencyInjection connection service used to read data
 	 */
-	public function update() -> boolean;
+	public function setReadConnectionService(string connectionService) -> void;
 
 	/**
-	 * Deletes a model instance. Returning true on success or false otherwise.
+	 * Sets the record's snapshot data.
+	 * This method is used internally to set snapshot data when the model was set up to keep snapshot data
+	 *
+	 * @param array columnMap
 	 */
-	public function delete() -> boolean;
+	public function setSnapshotData(array! data, columnMap = null);
 
 	/**
-	 * Returns the type of the latest operation performed by the ORM
-	 * Returns one of the OP_* class constants
+	 * Sets a transaction related to the Model instance
 	 */
-	public function getOperationMade() -> int;
+	public function setTransaction(<TransactionInterface> transaction) -> <ModelInterface>;
 
 	/**
-	 * Refreshes the model attributes re-querying the record from the database
+	 * Sets the DependencyInjection connection service used to write data
 	 */
-	public function refresh();
+	public function setWriteConnectionService(string connectionService) -> void;
 
 	/**
 	 * Skips the current operation forcing a success state
 	 */
 	public function skipOperation(boolean skip);
 
 	/**
-	 * Returns related records based on defined relations
+	 * Allows to calculate a sum on a column that match the specified conditions
 	 *
-	 * @param array arguments
+	 * @param array parameters
+	 * @return double
 	 */
-	public function getRelated(string alias, arguments = null) -> <ResultsetInterface>;
+	public static function sum(parameters = null);
```

Comment:
Return type?

Reply:
@virgofx That will come on the next pass. Still checking them. I found for instance that the `getEventsManager` was protected....